lib/stm32/f0/usart.c

```diff
@@ -31,399 +31,43 @@
-/** @brief USART Receiver Interrupt Disable.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- */
-
-void usart_disable_rx_interrupt(uint32_t usart)
-{
-	USART_CR1(usart) &= ~USART_CR1_RXNEIE;
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Transmitter Interrupt Enable.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- */
-
-void usart_enable_tx_interrupt(uint32_t usart)
-{
-	USART_CR1(usart) |= USART_CR1_TXEIE;
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Transmitter Interrupt Disable.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- */
-
-void usart_disable_tx_interrupt(uint32_t usart)
-{
-	USART_CR1(usart) &= ~USART_CR1_TXEIE;
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Error Interrupt Enable.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- */
-
-void usart_enable_error_interrupt(uint32_t usart)
-{
-	USART_CR3(usart) |= USART_CR3_EIE;
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Error Interrupt Disable.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- */
-
-void usart_disable_error_interrupt(uint32_t usart)
-{
-	USART_CR3(usart) &= ~USART_CR3_EIE;
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Read a Status Flag.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- * @param[in] flag Unsigned int32. Status register flag  @ref usart_sr_flags.
- * @returns boolean: flag set.
- */
-
-bool usart_get_flag(uint32_t usart, uint32_t flag)
-{
-	return ((USART_ISR(usart) & flag) != 0);
-}
-
-/*---------------------------------------------------------------------------*/
-/** @brief USART Return Interrupt Source.
- *
- * Returns true if the specified interrupt flag (IDLE, RXNE, TC, TXE or OE) was
- * set and the interrupt was enabled. If the specified flag is not an interrupt
- * flag, the function returns false.
- *
- * @param[in] usart unsigned 32 bit. USART block register address base @ref
- * usart_reg_base
- * @param[in] flag Unsigned int32. Status register flag  @ref usart_sr_flags.
- * @returns boolean: flag and interrupt enable both set.
- */
-
-bool usart_get_interrupt_source(uint32_t usart, uint32_t flag)
-{
-	uint32_t flag_set = (USART_ISR(usart) & flag);
-	/* IDLE, RXNE, TC, TXE interrupts */
-	if ((flag >= USART_ISR_IDLE) && (flag <= USART_ISR_TXE)) {
-		return ((flag_set & USART_CR1(usart)) != 0);
-	/* Overrun error */
-	} else if (flag == USART_ISR_ORE) {
-		return flag_set && (USART_CR3(usart) & USART_CR3_CTSIE);
-	}
-
-	return false;
-}
-
```

Comment:
This "diff" does not give a good representation of what happened to this file. It was all pulled out and put into the common module. The only thing in usart.c now is the get_clock() function which is unique to the F1 vs the L1 or L0.